src/pretix/presale/views/organizer.py

```diff
@@ -97,16 +97,24 @@ def get(self, request, *args, **kwargs):
             return super().get(request, *args, **kwargs)
 
     def get_queryset(self):
-        query = Q(is_public=True) & Q(live=True)
+        query = Q(is_public=True) & Q(live=True) & Q(organizer=self.request.organizer)
+        qs = Event.objects.filter(query)
         if "old" in self.request.GET:
-            query &= Q(Q(date_from__lte=now()) & Q(date_to__lte=now()))
+            qs = qs.annotate(
+                min_to=Min('subevents__date_to')
+            ).filter(
+                Q(Q(has_subevents=False) & Q(date_from__lte=now()) & Q(date_to__lte=now())) |
+                Q(Q(has_subevents=True) & Q(min_to__lte=now()))
+            )
             order = '-date_from'
         else:
-            query &= Q(Q(date_from__gte=now()) | Q(date_to__gte=now()))
+            qs = qs.annotate(
+                max_to=Max('subevents__date_to')
+            ).filter(
+                Q(Q(has_subevents=False) & Q(Q(date_from__gte=now()) | Q(date_to__gte=now()))) |
```

Comment:
Similar to above, the `date_from__gte=now()` check isn't really needed, as `date_to` should be `gte` anyway.